test/contrib/hsgp/test_approximation.py

```diff
@@ -154,8 +214,17 @@ def test_kernel_approx_squared_matern(
     eig_f1 = eigenfunctions(x1, ell=ell, m=m)
     eig_f2 = eigenfunctions(x2, ell=ell, m=m)
     approx = (eig_f1 * eig_f2) @ spd
-    exact = Matern(length_scale=length, nu=nu)(x1, x2)
-    assert jnp.isclose(approx, exact, rtol=1e-3)
+
+    def _exact_matern(length):
+        return Matern(length_scale=length, nu=nu)(x1, x2).squeeze(axis=-1)
+
+    if isinstance(length, float | int):
+        exact = _exact_matern(length)
+    elif length.ndim == 1:
+        exact = _exact_matern(length)
```

Comment:
why do you need two conditions for `exact = _exact_matern(length)`?

Reply:
(this is just a question, no need to use the OR statement in view of readability)

Reply:
I was copying the code from `test_kernel_approx_squared_exponential`. Yes, I believe this is in service to readability. If we use an 'or' statement here, we should probably use it there as well.

Reply:
either way fine by me